runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPlugin.java

```diff
@@ -164,6 +190,62 @@ public void onGameTick(GameTick gameTick)
 		}
 	}
 
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged varbitChanged)
+	{
+		updateVanillaHealthBarVars();
+	}
+
+	public void updateVanillaHealthBarVars()
+	{
+		clientThread.invokeLater(() ->
+		{
+			int opponentID = client.getVar(VarPlayer.HEALTH_BAR_OPPONENT);
+			int opponentCurrentHealth = client.getVar(Varbits.HEALTH_OVERLAY_BAR_CURRENT_VALUE);
+			int opponentMaximumHealth = client.getVar(Varbits.HEALTH_OVERLAY_BAR_MAX_VALUE);
+			opponentInfoOverlay.updateHealthBarVars(
+				opponentCurrentHealth,
+				opponentMaximumHealth ,
+				client.getNpcDefinition(opponentID).getName());
+		});
+	}
+
+	@Subscribe
+	public void onScriptPostFired(ScriptPostFired scriptPostFired)
+	{
+		if (scriptPostFired.getScriptId() == ScriptID.HEALTH_OVERLAY_BAR_UPDATING)
+		{
+			clientThread.invokeLater(() ->
+			{
+				opponentInfoOverlay.setVanillaHealthText();
+				opponentInfoOverlay.setHealthBarColors();
+			});
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("opponentinfo"))
+		{
+			if (event.getKey().equals("mergeVanillaHealthBar"))
+			{
+				if (event.getNewValue().equals("false"))
+				{
+					clientThread.invokeLater(() ->	opponentInfoOverlay.unHideVanillaHealthBar());
+				}
+			}
+			//reset/check values after config change.
+			clientThread.invokeLater(() ->
+			{
+				opponentInfoOverlay.setVanillaHealthText();
+				opponentInfoOverlay.setHealthBarColors();
+			});
+
+			updateVanillaHealthBarVars();
```

Comment:
Should these all be running even if mergeVanillaHealthBar wasn't the changed config within opponentinfo?